src/Compilers/CSharp/Test/Emit2/Semantics/LockTests.cs

```diff
@@ -0,0 +1,1797 @@
+            }
+
+            namespace System.Threading
+            {
+                public class Lock
+                {
+                    public Scope EnterLockScope()
+                    {
+                        Console.Write("E1 ");
+                        return new Scope();
+                    }
+
+                    public ref struct Scope
+                    {
+                        public void Dispose()
+                        {
+                            Console.Write("D1 ");
+                        }
+                    }
+                }
+            }
+            """;
+        var lib1 = CreateCompilation(source1)
+            .VerifyDiagnostics()
+            .EmitToImageReference();
+
+        var source2 = """
+            public static class C2
+            {
+                public static readonly System.Threading.Lock L = new();
+            }
+
+
+            namespace System.Threading
+            {
+                public class Lock
+                {
+                    public Scope EnterLockScope()
+                    {
+                        Console.Write("E2 ");
+                        return new Scope();
+                    }
+
+                    public ref struct Scope
+                    {
+                        public void Dispose()
+                        {
+                            Console.Write("D2 ");
+                        }
+                    }
+                }
+            }
+            """;
+        var lib2 = CreateCompilation(source2)
+            .VerifyDiagnostics()
+            .EmitToImageReference();
+
+        var source = """
+            using System;
+
+            static class Program
+            {
+                static void Main()
+                {
+                    M1();
+                    M2();
+                }
+
+                static void M1()
+                {
+                    var l1 = C1.L;
+                    lock (l1) { Console.Write("L1 "); }
+                }
+
+                static void M2()
+                {
+                    var l2 = C2.L;
+                    lock (l2) { Console.Write("L2 "); }
+                }
+            }
+            """;
+        var verifier = CompileAndVerify(source, [lib1, lib2], expectedOutput: "E1 L1 D1 E2 L2 D2");
+        verifier.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void LangVersion()
+    {
+        var source = """
+            using System;
+            using System.Threading;
+
+            Lock l = new Lock();
+            lock (l) { Console.Write("L"); }
+            """;
+
+        CSharpTestSource sources = [source, LockTypeDefinition];
+
+        CreateCompilation(sources, parseOptions: TestOptions.Regular12).VerifyDiagnostics(
+            // (5,7): error CS8652: The feature ''Lock' object' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
```

Comment:
>''Lock' object' [](http://example.com/codeflow?start=48&length=15)

The two apostrophes in a row look strange. Consider not surrounding "Lock" with them.